homeassistant/components/cover/device_trigger.py

```diff
@@ -0,0 +1,211 @@
+            )
+        if supported_features & SUPPORT_SET_POSITION:
+            triggers.append(
+                {
+                    CONF_PLATFORM: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "position",
+                }
+            )
+        if supported_features & SUPPORT_SET_TILT_POSITION:
+            triggers.append(
+                {
+                    CONF_PLATFORM: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "tilt_position",
+                }
+            )
+
+    return triggers
+
+
+async def async_get_trigger_capabilities(hass: HomeAssistant, config: dict) -> dict:
+    """List trigger capabilities."""
+    if config[CONF_TYPE] not in ["position", "tilt_position"]:
+        return {}
+
+    return {
+        "extra_fields": vol.Schema(
+            {
+                vol.Optional(CONF_ABOVE, default=0): vol.All(
+                    vol.Coerce(int), vol.Range(min=0, max=100)
+                ),
+                vol.Optional(CONF_BELOW, default=100): vol.All(
+                    vol.Coerce(int), vol.Range(min=0, max=100)
+                ),
+            }
+        )
+    }
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: AutomationActionType,
+    automation_info: dict,
+) -> CALLBACK_TYPE:
+    """Attach a trigger."""
+    config = TRIGGER_SCHEMA(config)
+
+    if config[CONF_TYPE] in STATE_TRIGGER_TYPES:
+        if config[CONF_TYPE] == "opened":
+            to_state = STATE_OPEN
+        elif config[CONF_TYPE] == "closed":
+            to_state = STATE_CLOSED
+        elif config[CONF_TYPE] == "opening":
+            to_state = STATE_OPENING
+        elif config[CONF_TYPE] == "closing":
+            to_state = STATE_CLOSING
+
+        state_config = {
+            state_automation.CONF_PLATFORM: "state",
+            CONF_ENTITY_ID: config[CONF_ENTITY_ID],
+            state_automation.CONF_TO: to_state,
+        }
+        state_config = state_automation.TRIGGER_SCHEMA(state_config)
+        return await state_automation.async_attach_trigger(
+            hass, state_config, action, automation_info, platform_type="device"
+        )
+
+    if config[CONF_TYPE] == "position":
+        position = "current_position"
+    if config[CONF_TYPE] == "tilt_position":
+        position = "current_tilt_position"
+    min_pos = config.get(CONF_ABOVE, -1)
+    max_pos = config.get(CONF_BELOW, 101)
+    value_template = (
+        f"{{{{ (state_attr('{config[ATTR_ENTITY_ID]}', '{position}')|int) > {min_pos}"
+        f"and (state_attr('{config[ATTR_ENTITY_ID]}', '{position}')|int) < {max_pos} }}}}"
+    )
+
+    template_config = {
+        template_automation.CONF_PLATFORM: "template",
```

Comment:
We should not use the template trigger for this. Use the `numeric_state` trigger.

Reply:
Fixed.